DataFormats/TauReco/interface/RecoTauPiZero.h

```diff
@@ -14,32 +14,57 @@ class RecoTauPiZero : public CompositePtrCandidate {
       kStrips = 3
     };
 
-    RecoTauPiZero():CompositePtrCandidate(),algoName_(kUndefined){
-      this->setPdgId(111); }
-
-    RecoTauPiZero(PiZeroAlgorithm algoName):
-        CompositePtrCandidate(), algoName_(algoName) { this->setPdgId(111); }
+    RecoTauPiZero()
+      : CompositePtrCandidate(),
+        algoName_(kUndefined)
```

Comment:
sigh.
``` CompositePtrCandidate(),    algoName_(kUndefined), bendCorrEta_ (0.), bendCorrPhi_( 0.) ```

May I suggest to leaf through an entry level book on c++ programming  before working on a next PR?